test/functional/wallet_pruning.py

```diff
@@ -0,0 +1,132 @@
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""Test wallet import on pruned node."""
+import os
+
+from test_framework.util import assert_raises_rpc_error
+from test_framework.blocktools import create_block
+from test_framework.blocktools import create_coinbase
+from test_framework.test_framework import BitcoinTestFramework
+
+from test_framework.script import (
+    CScript,
+    OP_RETURN,
+    OP_TRUE,
+)
+
+class WalletPruningTest(BitcoinTestFramework):
+    nTime = 0
+    def mine_large_blocks(self, node, n):
+        # Get the block parameters for the first block
+        best_block = node.getblock(node.getbestblockhash())
+        height = int(best_block["height"]) + 1
+        self.nTime = max(self.nTime, int(best_block["time"])) + 1
+        previousblockhash = int(best_block["hash"], 16)
+        big_script = CScript([OP_RETURN] + [OP_TRUE] * 950000)
+        for _ in range(n):
+            block = create_block(hashprev=previousblockhash, ntime=self.nTime, coinbase=create_coinbase(height, script_pubkey=big_script))
+            block.solve()
+
+            # Submit to the node
+            node.submitblock(block.serialize().hex())
+
+            previousblockhash = block.sha256
+            height += 1
+
+            # Simulate 10 minutes of work time per block
+            # Important for matching a timestamp with a block +- some window
+            self.nTime += 600
+            for n in self.nodes:
+                if n.running:
+                    n.setmocktime(self.nTime) # Update node's time to accept future blocks
+
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [
+            [], # node dedicated to mining
+            ['-prune=550'], # node dedicated to testing pruning
+        ]
+
+    def setup_nodes(self):
+        self.add_nodes(self.num_nodes, extra_args=self.extra_args)
+        self.start_node(0)
+
+    def setup_network(self):
+        self.setup_nodes()
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def create_big_chain(self):
+        self.log.info("Generating a long chain of blocks...")
+        # Generate 288 light blocks, the minimum required to stay on disk with prune enabled
+        # do not sync
+        self.generate(self.nodes[0], 288, sync_fun=lambda: None)
+
+        # Generate large blocks to make sure we have enough to test chain pruning
+        self.mine_large_blocks(self.nodes[0], 600)
+
+    def test_wallet_import_pruned_test(self):
+        self.log.info("Make sure we can import wallet when pruned and required blocks are still available")
+
+        wname = "wallet_pruned.dat"
+
+        # export wallet
+        self.nodes[1].dumpwallet(os.path.join(self.nodes[1].datadir, wname))
+        # import wallet
+        self.nodes[1].importwallet(os.path.join(self.nodes[1].datadir, wname))
```

Comment:
This dump&import pattern in the same wallet might be ok for checking against the RPC assertion of the newly introduced logic, but in reality the keys and scripts in the dump file are not imported in the wallet because they already exist. 

I don't consider this as an issue because that behavior is also tested elsewhere(`wallet_dump.py`) but I think changing this part makes for a better usage scenario.